logstash-core/src/main/java/org/logstash/FileLockFactory.java

```diff
@@ -39,7 +41,8 @@
 
     private FileLockFactory() {}
 
-    private static final Set<String> LOCK_HELD = Collections.synchronizedSet(new HashSet<String>());
+    private static final Set<String> LOCK_HELD = Collections.synchronizedSet(new HashSet<>());
```

Comment:
I guess I am asking for the context of that other branch. Without that context, I can't really say whether I like the usage of both `LOCK_HELD` and `LOCK_MAP`. For example, maybe something can change as to not need both?

If that is not the case, and it is just a matter of fact that some places only have the `Lock`, while others have the `filePath`, I understand.